### PR TITLE
Refactoring configure block handling.

### PR DESCRIPTION
**Pull request by hekaldama**
This gives us the ability to use plain ruby datastructures (Array, Hash,
and String) to try and describe the syntax of the configure block. It
treats the whole configure block like an Array of lines and iterates
through just printing the lines faithfully.

### CODE REVIEW

File: jenkins-xml-to-jobdsl.rb [hunk 1]
```diff
@@ -620,17 +620,17 @@ def process(job_name, depth, indent)
       when 'notifyCulprits'
         puts " " * currentDepth + "notifyScmCulprits(#{i.text})"
       when 'notifyOnBuildStart'
-        configureBlock << "'#{i.name}'(#{i.text})"
+        configureBlock << "(ircNode / '#{i.name}').setValue(#{i.text})"
```
Comment (visualphoenix) [inline jenkins-xml-to-jobdsl.rb]: this is odd - i dont see ircNode used

Comment (visualphoenix) [inline jenkins-xml-to-jobdsl.rb]: Can you make the PR only about the configure block?

Comment (hekaldama) [inline jenkins-xml-to-jobdsl.rb]: `ircNode` is used on line 632

Comment (hekaldama) [inline jenkins-xml-to-jobdsl.rb]: > Can you make the PR only about the configure block?
I can understand why you would want this, but would rather go over it in detail tomorrow if you have time and then still see if you would like a change to the PR.

**Comment by visualphoenix** [comment]
Sounds good. Let's chat tomorrow.

**Review by visualphoenix — APPROVED** [review]
Had a chance to more thoroughly review this. It makes sense.

**Pull request merged**